### PR TITLE
JTFS frontend: remove average, add Q_fr and J_fr

### PR DESCRIPTION
**Pull request by cyrusasfa**

### CODE REVIEW

File: kymatio/scattering1d/frontend/torch_frontend.py [hunk 1]
```diff
@@ -145,34 +145,35 @@ def scattering(self, x):
...
+                                   oversampling=oversampling, out_type=out_type, 
+                                   backend=backend)
 
         # First-order scattering object for the frequency variable
-        max_order_fr = 1
         shape_fr = (Q * J)
-        J_fr = self.get_J_fr()
-        Q_fr = 1
+        self.J_fr = self.get_J_fr() if not J_fr else J_fr
+        self.Q_fr = Q_fr
```
Comment (MuawizChaudhary) [inline kymatio/scattering1d/frontend/torch_frontend.py:171]: isn't `Q_fr` supposed to be equal to 1 in JTFS?

File: kymatio/scattering1d/frontend/torch_frontend.py [hunk 2]
```diff
@@ -145,34 +145,35 @@ def scattering(self, x):
...
-            oversampling=oversampling, vectorize=vectorize, out_type=out_type, 
-            backend=backend)
+        ScatteringTorch1D.__init__(self, J, shape, 
+                                   Q=Q, T=T, max_order=max_order_tm,
+                                   oversampling=oversampling, out_type=out_type, 
+                                   backend=backend)
 
         # First-order scattering object for the frequency variable
-        max_order_fr = 1
         shape_fr = (Q * J)
```
Comment (MuawizChaudhary) [inline kymatio/scattering1d/frontend/torch_frontend.py]: do we wanna save the shape?

Comment (MuawizChaudhary) [reply kymatio/scattering1d/frontend/torch_frontend.py]: `Q * J ` is a lower bound. Number of wavelets, but also have to consider the evenly spaced filters. (which are in the low frequency)

File: kymatio/scattering1d/frontend/torch_frontend.py (hunk 1 above)
Comment (MuawizChaudhary) [reply kymatio/scattering1d/frontend/torch_frontend.py:171]: "Not necessarily."

File: kymatio/scattering1d/frontend/torch_frontend.py [hunk 3]
```diff
@@ -199,31 +200,19 @@ def scattering(self, x):
...
-            self.J,
-            self.psi1_f, self.psi2_f, self.phi_f,
-            self.sc_freq,
-            average=self.average,
-            pad_left=self.pad_left, pad_right=self.pad_right,
-            ind_start=self.ind_start, ind_end=self.ind_end,
-            oversampling=self.oversampling,
-            size_scattering=size_scattering,
-            out_type=self.out_type)
+        S = timefrequency_scattering(x, self.backend.pad, self.backend.unpad,
```
Comment (MuawizChaudhary) [inline kymatio/scattering1d/frontend/torch_frontend.py:203]: supposed to pass `self.J_fr` to this function.

**Comment by cyrusasfa** [comment]
merge?

**Pull request merged**